packages/bitcore-node/test/integration/models/coin.spec.ts

```diff
@@ -1,7 +1,230 @@
+      }
     });
-});
\ No newline at end of file
+
+    await TransactionStorage.pruneMempool({
+      chain,
+      network,
+      initialSyncComplete: true,
+      spendOps: spentOps
+    });
+
+    const badTxs = await TransactionStorage.collection.find({ chain, network, txid: { $in: txids } }).toArray();
+    expect(badTxs.length).to.eq(chainLength);
+    expect(badTxs.map(tx => tx.blockHeight)).to.deep.eq(new Array(chainLength).fill(SpentHeightIndicators.conflicting));
+
+    const goodTxs = await TransactionStorage.collection.find({ chain, network, txid: blockTx.txid }).toArray();
+    expect(goodTxs.length).to.eq(1);
+    expect(goodTxs[0].txid).to.eq(blockTx.txid);
+    expect(goodTxs[0].blockHeight).to.eq(blockTx.blockHeight);
+
+    // Coins
+    const badNewCoins = await CoinStorage.collection.find({ chain, network, mintTxid: { $in: txids } }).toArray();
+    expect(badNewCoins.length).to.equal(badNewCoins.filter(c => c.spentHeight == SpentHeightIndicators.pending).length);
+
+    const goodNewCoins = await CoinStorage.collection.find({ chain, network, mintTxid: blockTx.txid }).toArray();
+    expect(goodNewCoins.length).to.equal(
+      goodNewCoins.filter(c => c.spentHeight == SpentHeightIndicators.unspent).length
+    );
+  });
+
+  it('should appropriately mark coins related to transactions that are RBFed', async () => {
+    const privateKey = new BitcoreLib.PrivateKey('L1uyy5qTuGrVXrmrsvHWHgVzW9kKdrp27wBC7Vs6nZDTF2BRUVwy');
+
+    const utxo1 = {
+      txId: createNewTxid(),
+      outputIndex: 0,
+      address: '17XBj6iFEsf8kzDMGQk5ghZipxX49VXuaV',
+      script: '76a91447862fe165e6121af80d5dde1ecb478ed170565b88ac',
+      satoshis: 50000
+    };
+
+    // create tx with mutliple outputs
+    const tx1 = new BitcoreLib.Transaction()
+      .from(utxo1)
+      .to('1Gokm82v6DmtwKEB8AiVhm82hyFSsEvBDK', 15000)
+      .to('1Gokm82v6DmtwKEB8AiVhm82hyFSsEvBDK', 13000)
+      .to('1Gokm82v6DmtwKEB8AiVhm82hyFSsEvBDK', 11000)
+      .sign(privateKey);
+
+    // import transaction in block 1
+    await TransactionStorage.batchImport({
+      txs: [tx1],
+      height: 1,
+      initialSyncComplete: true,
+      chain: 'BTC',
+      network: 'integration'
+    });
+
+    // insert mempool tx using all outputs from last tx
+    const mempoolTxid = createNewTxid();
+    const mempoolTx = {
+      chain,
+      network,
+      blockHeight: -1, // pending
+      txid: mempoolTxid
+    } as IBtcTransaction;
+    const mempoolOutputs = Array.from({ length: 3 }, (_v, i) => i).map(i => {
+      return {
+        chain,
+        network,
+        mintHeight: -1,
+        mintTxid: mempoolTxid,
+        mintIndex: i,
+        spentHeight: SpentHeightIndicators.unspent
+      } as ICoin;
+    });
+    await addTx(mempoolTx, mempoolOutputs);
+    
+    // update existing outputs to be spent by mempool tx
+    CoinStorage.collection.updateMany(
```

Comment:
Need to await?

Reply:
Good catch. Added await in the latest commit.